caluma/core/validations.py

```diff
@@ -0,0 +1,66 @@
+import inspect
+from functools import wraps
+
+from django.core.exceptions import ImproperlyConfigured
+
+from .collections import list_duplicates
+
+
+def validation_for(mutation):
+    """Decorate function to overwriting validation of specific mutation."""
+
+    def decorate(fn):
+        @wraps(fn)
+        def _decorated(self, mutation, data, info):
+            return fn(self, mutation, data, info)
+
+        _decorated._validation = mutation
+        return _decorated
+
+    return decorate
+
+
+class BaseValidation(object):
+    """Basic validation class to be extended by any validation implementation.
+
+    In combination with the decorators `@validation_for` a custom validation class
+    can define validation on basis of mutations and its subclasses.
+
+    Per default it doesn't do any validation but default can be overwritten to define
+    a validation for `Mutation`.
```

Comment:
(included in suggestion above)

Reply:
ahh I see what you we trying to do with above comment... should review first all comments before recommenting... :smile: let me try to rephrase this

Reply:
:smile: